# Homogeneous system

a system of linear equations is homogeneous if its augmented column is 0 vector,that is Ax=0.it always has zero vector as a solution wich

is trivial solution,we want to see if it has non trivial solution,that amounts to free variables.we can use it to create parameteric

vector form of solutions of system of linear equations.if vh is solution set of Ax=0 and p is solution of of Ax = b then each w=vh+p

is solution of AX=b since Aw = A(vh + P) = Avh + Ap = 0 + b = b.constructing parametric vector form

In [ ]:
#Importing numpy
import numpy as np

In [19]:
#Coefficent matrix and augmented column
coefficent_matrix =  np.array([[3 , 5 , -4] , [-3 , -2 , 4] , [6 , 1 ,-8]] , dtype=np.float64)

augmented_column = np.array([[7 , -1 , -4]] , dtype=np.float64)

print(coefficent_matrix)

print()

print(augmented_column)

[[ 3.  5. -4.]
 [-3. -2.  4.]
 [ 6.  1. -8.]]

[[ 7. -1. -4.]]


In [34]:
#Augmented matrix
augmented_matrix = np.hstack((coefficent_matrix , augmented_column.T))

print(augmented_matrix)

[[ 3.  5. -4.  7.]
 [-3. -2.  4. -1.]
 [ 6.  1. -8. -4.]]


In [35]:
#Reduced row echelon form
augmented_matrix[1] += augmented_matrix[0]

print(augmented_matrix)

print()

augmented_matrix[2] -= (2 * augmented_matrix[0])

print(augmented_matrix)

print()

augmented_matrix[2] += (3* augmented_matrix[1])

print(augmented_matrix)

print()

augmented_matrix[1] /= 3

print(augmented_matrix)

print()

augmented_matrix[0] -= (5 * augmented_matrix[1])

print(augmented_matrix)

print()

augmented_matrix[0] /= 3

print(augmented_matrix)

[[ 3.  5. -4.  7.]
 [ 0.  3.  0.  6.]
 [ 6.  1. -8. -4.]]

[[  3.   5.  -4.   7.]
 [  0.   3.   0.   6.]
 [  0.  -9.   0. -18.]]

[[ 3.  5. -4.  7.]
 [ 0.  3.  0.  6.]
 [ 0.  0.  0.  0.]]

[[ 3.  5. -4.  7.]
 [ 0.  1.  0.  2.]
 [ 0.  0.  0.  0.]]

[[ 3.  0. -4. -3.]
 [ 0.  1.  0.  2.]
 [ 0.  0.  0.  0.]]

[[ 1.          0.         -1.33333333 -1.        ]
 [ 0.          1.          0.          2.        ]
 [ 0.          0.          0.          0.        ]]


In [41]:
#Parametric vector form
print("Parametrix vector form: ")

print(f"{augmented_matrix[: , 3]} + x3 * {[-1.33333 , 0 , 1]}")

Parametrix vector form: 
[-1.  2.  0.] + x3 * [-1.33333, 0, 1]


As we see,[-1.33333 , 0 , 1] is solution of AX=0 ,the solution indicates answer is a line through [1 , 2 , 0] parallel to 

[-1.333333 , 0 , 1].generally solutions can be descirbed as a hyperplane through constant parallel to homogeneous solution span.